### ISSUE TITLE
Backing property requires other value to be part of public API

### ISSUE THREAD

**Issue by paul-dingemans** [pinterest/ktlint#2345]
[According to Kotlin Coding conventions](https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties):
> If a class has two properties which are conceptually the same but one is part of a public API and another is an implementation detail, use an underscore as the prefix for the name of the private property:

Currently it is not checked that the variable not starting with `_` is a part of the `public` API. As a result the `_elementList` property is code below is not reported:
```
            class Foo {
                private val _elementList = mutableListOf<Element>()

                private val elementList: List<Element>
                    get() = _elementList
            }
```

**Comment by Goooler** [comment]
We can ignore this `elementList` if it's `public` or `internal`?

**Comment by paul-dingemans** [comment]
> We can ignore this `elementList` if it's `public` or `internal`?

I am not sure whether I understand your question. As I read the documentation, a backing property requires a `private` property of which the name starts with `_` and a `public` property with same name but not starting with `_`.

**Comment by Goooler** [comment]
Yes, I meant if an `internal` property with same name but not starting with `_`, should we also ignore this case?

**Comment by paul-dingemans** [comment]
If we go this route, more exceptions will follow. The end result will be that properties can always start with an `_`. I think we should as close to the coding conventions as is possible. We can consider to split this rule and move cheking of the backing property to a separate rule, which then can be disabled if a project uses backing properties less strict than the coding conventions.
